### ISSUE TITLE
Non-literal list concatenation and sorting: TypeError: t.py_sort is not a function

### ISSUE THREAD

**Issue by darthglowball** [TranscryptOrg/Transcrypt#679]
Hi,

sorting a non-literal concatenated list causes the error: TypeError: t.py_sort is not a function. While the lists are literals, the concatenation is done by variables and the plus operator.

```
a = [5,6,7]
b = [8,9,10]
t = a + b
t.sort()
print(t)
```

**Comment by vkbsb** [comment]
have you tried t.js_sort() instead?

**Comment by darthglowball** [comment]
@vkbsb no, thanks, but I fixed it with some after-processing on the Javascript by changing `a + b` to `a.append(b)` while keeping py_sort. I want to keep my code as Pythonic as possible, so I think using a js_sort would defeat that purpose (maybe not for this example, but for others).

**Comment by fzzylogic** [comment]
@darthglowball You may want to [enabe operator overloading](https://transcrypt.org/numscrypt/docs/html/special_facilities.html#operator-overloading-pragma-opov-and-pragma-noopov) when compiling (using -o switch for global, or __pragma__('opov') for local).

**Comment by darthglowball** [comment]
@fzzylogic something happens now, but in Firefox I get the incorrect result `[10, 5, 6, 7, 8, 9]`.

**Comment by darthglowball** [comment]
Even with the -o switch, you get a py_sort in the generated code:

```
(function () {
	var __accu0__ = t;
	return __call__ (__accu0__.py_sort, __accu0__);
}) ();
```

That doesn't behave like a Python sort (see previous post).

**Comment by AlexECX** [comment]
In the transcrypt runtime.js the sort function used is the javascript `.sort()` (`__sort__` function, Line:1976). The problem is that, in javascript, `.sort()` treats strings and/or numbers as strings and sort alphabetically. To correct this, Line:1976 should be changed from:
`iterable.sort ();`
to:
`iterable.sort ((a,b) => a > b ? 1 : -(1));`

Even better would be to change Line:1972 and Line:1976 to use `__gt__(a,b)` instead of `a > b`.

@darthglowball incidentally, `.py_sort` will branch to a code section that does exactly what this fix will do when using the `key` parameter of the python `.sort()`. Using `t.sort(lambda a:a)` will sort correctly.

**Comment by AlexECX** [comment]
related to #605

**Comment by AlexECX** [comment]
More precisely in [`transcrypt/modules/org/transcrypt/__runtime__.py`](https://github.com/QQuick/Transcrypt/blob/24fa83096cf3335e50b4f1a49ada3578c090126c/transcrypt/modules/org/transcrypt/__runtime__.py#L107):

`107|        iterable.sort ()                                            # JavaScript sort`
could be :
`107|        iterable.sort (lambda a, b: 1 if a > b else -1)`

Output:
```
> [10,1,2,20,9].sort()
 [1, 2, 9, 10, 20]

> ["10","1","2","20","9"].sort()
["1", "10", "2", "20", "9"]
```

**Comment by AlexECX** [comment]
> @AlexECX yes, Python `t.sort()` maps to Javascript `t.sort(function(a,b){return a>b})` but while Transcrypt allows a positional or keyword argument in sort, it should be a keyword argument in Python 2.7.16 & 3.7.2.

Currently Transcrypts `.sort()` maps to Javascript's `.sort()`, which is why it sorts alphabetically. Mapping it to `.sort ((a,b) => a > b ? 1 : -(1))` instead will resolve the sorting order. (Note that `.sort(function(a,b){return a>b})` has the same problem, at least in chrome).

I believe the Transcrypt `sort()` signature is practically the same as Python's (key = None, reverse = False), but currently the arguments are not passed properly (see #687).

**Comment by darthglowball** [comment]
@AlexECX you're right about Chrome giving the incorrect result with `.sort(function(a,b){return a>b})`. Same problem with Opera. Firefox does it correctly. `.sort ((a,b) => a > b ? 1 : -(1))` works on all three. Good catch.

**Comment by JennaSys** [comment]
> More precisely in [`transcrypt/modules/org/transcrypt/__runtime__.py`](https://github.com/QQuick/Transcrypt/blob/24fa83096cf3335e50b4f1a49ada3578c090126c/transcrypt/modules/org/transcrypt/__runtime__.py#L107):
> 
> `107| iterable.sort () # JavaScript sort` could be : `107| iterable.sort (lambda a, b: 1 if a > b else -1)`

This has been implemented in the dev_3.9.3 branch to be released in v3.9.3

**Comment by JennaSys** [comment]
Merged to master on PR #876
